From 34a70a8041862757b951fb25fa060386c9c7db0e Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Tue, 14 Mar 2017 22:39:20 -0400
Subject: [PATCH 74/75] Add s390/s390x set-default-index test (#1285601)

Adds a test to ensure set-default-index actually works for s390 and
s390x platforms.

Related: rhbz#1285601
---
 test.sh                   |  2 ++
 test/results/default/z2.1 | 22 ++++++++++++++++++++++
 2 files changed, 24 insertions(+)
 create mode 100644 test/results/default/z2.1

diff --git a/test.sh b/test.sh
index 34ba4d7..7225c9f 100755
--- a/test.sh
+++ b/test.sh
@@ -314,6 +314,8 @@ liloDisplayTest lilo.1 defaultkernel/l1.1 --default-kernel
 testing="Z/IPL default directive"
 ziplTest zipl.1 default/z1.1 --add-kernel /boot/new-kernel --title test
 ziplTest zipl.1 default/z1.2 --add-kernel /boot/new-kernel --title test --make-default
+testing="Z/IPL default index directive"
+ziplTest zipl.2 default/z2.1 --set-default-index=0
 testing="Z/IPL display default index"
 ziplDisplayTest zipl.1 defaultindex/0 --default-index
 testing="Z/IPL display default title"
diff --git a/test/results/default/z2.1 b/test/results/default/z2.1
new file mode 100644
index 0000000..93e7491
--- /dev/null
+++ b/test/results/default/z2.1
@@ -0,0 +1,22 @@
+[defaultboot]
+defaultauto
+prompt=1
+timeout=5
+default=3.10.0-514.6.2.el7.s390x
+target=/boot
+[3.10.0-514.6.2.el7.s390x]
+	image=/boot/vmlinuz-3.10.0-514.6.2.el7.s390x
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8"
+	ramdisk=/boot/initramfs-3.10.0-514.6.2.el7.s390x.img
+[3.10.0-514.6.2.el7.s390x_with_debugging]
+	image=/boot/vmlinuz-3.10.0-514.6.2.el7.s390x
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8 systemd.log_level=debug systemd.log_target=kmsg"
+	ramdisk=/boot/initramfs-3.10.0-514.6.2.el7.s390x.img
+[linux]
+	image=/boot/vmlinuz-3.10.0-514.el7.s390x
+	ramdisk=/boot/initramfs-3.10.0-514.el7.s390x.img
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0 LANG=en_US.UTF-8"
+[linux-0-rescue-196fd0394b7442a6838686745f03fd04]
+	image=/boot/vmlinuz-0-rescue-196fd0394b7442a6838686745f03fd04
+	ramdisk=/boot/initramfs-0-rescue-196fd0394b7442a6838686745f03fd04.img
+	parameters="root=/dev/mapper/rhel_ibm--z--68-root crashkernel=auto rd.dasd=0.0.0120 rd.dasd=0.0.0121 rd.dasd=0.0.0122 rd.dasd=0.0.0123 rd.lvm.lv=rhel_ibm-z-68/root rd.lvm.lv=rhel_ibm-z-68/swap rd.znet=qeth,0.0.8000,0.0.8001,0.0.8002,layer2=1,portname=z-68,portno=0"
-- 
2.9.3

